SignalServiceKit/src/Storage/FullTextSearchFinder.swift

```diff
@@ -170,6 +209,12 @@ public class FullTextSearchFinder: NSObject {
     }
 
     private class var dbExtensionConfig: YapDatabaseFullTextSearch {
+        SwiftAssertIsOnMainThread(#function)
+
+        if dbConnection == nil {
+            dbConnection = OWSPrimaryStorage.shared().newDatabaseConnection()
```

Comment:
You might not like this; I'm trying to ensure that the db connection gets initialized exactly once, on the main thread, and at a well-defined time.

Reply:
This is indeed a strange place for a side effect like this. Why not use the global read connection?

Reply:
Removed.  I thought you would prefer this way.